codegen/s3.rs

```diff
@@ -11786,22 +11657,37 @@ impl<'a> S3Client<'a> {
 	}
 	/// Completes a multipart upload by assembling previously uploaded parts.
 	pub fn complete_multipart_upload(&mut self, input: &CompleteMultipartUploadRequest) -> Result<CompleteMultipartUploadOutput, AWSError> {
-		let mut request = SignedRequest::new("POST", "s3", &self.region, "/{Bucket}/{Key+}");
+		let mut request = SignedRequest::new("POST", "s3", &self.region,
+			&format!("/{}", input.key));
+
 		let mut params = Params::new();
-		params.put("Action", "CompleteMultipartUpload");
-		CompleteMultipartUploadRequestWriter::write_params(&mut params, "", &input);
+		params.put("uploadId", &input.upload_id.to_string());
 		request.set_params(params);
-		let result = request.sign_and_execute(&self.creds.get_credentials());
+
+		let hostname = (&input.bucket).to_string() + ".s3.amazonaws.com";
+		request.set_hostname(Some(hostname));
+
+		request.set_payload(input.multipart_upload);
+
+		let mut result = request.sign_and_execute(&self.creds.get_credentials());
 		let status = result.status.to_u16();
-		let mut reader = EventReader::new(result);
-		let mut stack = XmlResponseFromAws::new(reader.events().peekable());
-		stack.next(); // xml start tag
-		stack.next();
+
+		println!("status is {}", status);
+
 		match status {
 			200 => {
-				Ok(try!(CompleteMultipartUploadOutputParser::parse_xml("CompleteMultipartUploadOutput", &mut stack)))
+				let mut reader = EventReader::new(result);
```

Comment:
This is a much better pattern to use than doing it outside the matcher.  We should switch things to this instead of parsing outside the match block.